web-gateway/app/controllers/Main.java

```diff
@@ -55,6 +56,7 @@ public Main(Configuration config, MessagesApi messagesApi, UserService userServi
                     }
 
                     return userService.createUser().invoke(new User(form.get().getName())).thenApply(user -> {
+                        userService.updateCredential().invoke(new Credential(user.getId(), form.get().getUsername(), form.get().getPassword()));
```

Comment:
Consider that there could be a failure in between when `createUser` is invoked and when `updateCredential` is invoked. For example, the server running this code could crash, or the network between the Play app and the user service could go down in between calls. That would leave the user in a half-created state: they wouldn't be able to log in.